### ISSUE TITLE
Logsigmoid as chaining operation

### ISSUE THREAD

**Issue by Bartolo1024** [pytorch/pytorch#42597]
## 🚀 Feature

I would like to have a chained log sigmoid operation in `torch.Tensor` object.

## Motivation

I prefer to use chained versions of the operations like `x.sum()`, `x.mean()`, etc. instead of `torch.sum()` `torch.mean()`, etc.
This prevents me from the formulation of hardly readable long operations.
For example:
```x.pow(2).sum(dim=1).mean()```
is much cleaner than:
```torch.mean(torch.sum(torch.pow(x, 2), dim=1))```

Currently, the operation is available only in torch.nn or as torch module method or can be implemented as the combination of sigmoid and logarithm, which is less numerically stable.

## Pitch

Add log sigmoid to tensor object methods.
```
import torch
x = torch.rand(2, 3, 10, 10)
log_sigmoid_from_x = x.logsigmoid()
```

**Comment by malfet** [comment]
Hmm, `sigmoid`  is method of `Tensor`, so why not `logsigmoid` couldn't be one.
@ezyang, what do you think?

**Comment by ezyang** [comment]
...suuuuuure? it's one darn slippery slope, but logsigmoid might deserve it. cc @gchanan

**Comment by vadimkantorov** [comment]
softmax/logsumexp is also very-very common, may deserve migration from torch.nn.functional to torch, but it's a slippery slope indeed...

**Comment by Bartolo1024** [comment]
@vadimkantorov softmax/logsumexp/log_softmax are already available as torch module and torch.Tensor methods?

**Comment by vadimkantorov** [comment]
@Bartolo1024 I think this is currently unintetional...

**Comment by firstprayer** [comment]
Only place I found so far related to how is Tensor.xxx() implemented is this

https://github.com/pytorch/pytorch/blob/61623430d32c081124067fc4abf33bd4a023af90/torch/csrc/jit/runtime/symbolic_script.cpp#L783-L788

Looks like it wraps it within a helper function but still calling torch.xxx() under the hook>
